synapseclient/client.py

```diff
@@ -621,6 +621,71 @@ def invalidateAPIKey(self):
         if self._is_logged_in():
             self.restDELETE("/secretKey", endpoint=self.authEndpoint)
 
+    @functools.lru_cache()
+    def get_user_profile_by_username(
+        self,
+        username: str = None,
+        sessionToken: str = None,
+    ) -> UserProfile:
+        """
+        Get the details about a Synapse user.
+        Retrieves information on the current user if 'id' is omitted or is empty string.
+        :param username:           The userName of a user
+        :param sessionToken: The session token to use to find the user profile
+        :returns: The user profile for the user of interest.
+
+        Example::
+            my_profile = syn.get_user_name_profile()
+            freds_profile = syn.get_user_name_profile('fredcommo')
+        """
+        if not isinstance(username, str) and username is not None:
```

Comment:
nit: It would read a little better to assign these conditions to variables like:

is_none = username is None
is_str = isinstance(username, str)
if not is_str and not is_none:
    # type error
if is_str:
    # do stuff

Reply:
added in latest commit